contracts/src/contracts/nullifier_registry/nullifier_registry.cairo

```diff
@@ -1 +1,294 @@
+#[starknet::contract]
+pub mod NullifierRegistry {
+    use starknet::{
+        ContractAddress, get_caller_address, get_block_timestamp,
+        storage::{Map, StorageMapReadAccess, StorageMapWriteAccess}
+    };
+    use zkramp::contracts::nullifier_registry::interface::{
+        INullifierRegistry, INullifierRegistryDispatcher, INullifierRegistryDispatcherTrait
+    };
+
+    #[storage]
+    pub struct Storage {
+        // / map(index -> writers address)
+        writers: Map<u64, ContractAddress>,
+        writers_count: u64,
+        is_nullified: Map<u256, bool>,
+        is_writers: Map<ContractAddress, bool>
+    }
+
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    pub enum Event {
+        NullifierAdded: NullifierAdded,
+        WriterAdded: WriterAdded,
+        WriterRemoved: WriterRemoved
+    }
+
+    /// @notice Emitted when the account is locked
```

Comment:
Docstring seems to be from some different function?

Reply:
woo, i will remove it